astropy/timeseries/binned.py

```diff
@@ -119,6 +119,13 @@ def __init__(self, data=None, *, time_bin_start=None, time_bin_end=None,
         if isinstance(time_bin_size, TimeDelta):
             time_bin_size = time_bin_size.sec * u.s
 
+        # If any one of the parameters 'time_bin_start' or 'time_bin_end' is not scalar,
+        # an exception is raised.
+        if n_bins is not None and time_bin_end is not None:
```

Comment:
Docstring at https://docs.astropy.org/en/latest/api/astropy.timeseries.BinnedTimeSeries.html does, "This is only used if both `time_bin_start` and `time_bin_size` are provided and are scalar values." The problem here is that you are checking `time_bin_end` instead of `time_bin_size`. Hopefully this solves the failing test case.

```suggestion
        if n_bins is not None and time_bin_size is not None:
            if not (time_bin_start.isscalar and  time_bin_size.isscalar):
```